keras/utils/visualize_util.py

```diff
@@ -64,4 +66,9 @@ def model_to_dot(model, show_shapes=False, show_layer_names=True):
 
 def plot(model, to_file='model.png', show_shapes=False, show_layer_names=True):
     dot = model_to_dot(model, show_shapes, show_layer_names)
-    dot.write_png(to_file)
+    _, format = os.path.splitext(to_file)
+    if not format:
+        format = 'png'
+    else:
+        format = format[1:]
```

Comment:
I believe it's quite difficult to check that, since pydot doesn't provide a proper api, and dot program also check this at runtime. On the other hand, when format is not supported, the error message from dot is fairly clear.